src/hotspot/cpu/riscv/c2_MacroAssembler_riscv.cpp

```diff
@@ -2134,30 +2134,77 @@ void C2_MacroAssembler::enc_cmove(int cmpFlag, Register op1, Register op2, Regis
 
 // Set dst to NaN if any NaN input.
 void C2_MacroAssembler::minmax_fp(FloatRegister dst, FloatRegister src1, FloatRegister src2,
-                                  bool is_double, bool is_min) {
-  assert_different_registers(dst, src1);
-  assert_different_registers(dst, src2);
+                                  FLOAT_TYPE ft, bool is_min) {
+  assert_different_registers(dst, src1, src2);
+  if (ft == FLOAT_TYPE::half_precision) {
+    assert_cond(UseZfh);
+  }
```

Comment:
Suggestion: `assert_cond((ft != FLOAT_TYPE::half_precision) || UseZfh);`

Reply:
OK, will fix it.